lib/_stream_readable.js

```diff
@@ -338,11 +338,9 @@ Readable.prototype.setEncoding = function(enc) {
   this._readableState.encoding = this._readableState.decoder.encoding;
 
   // Iterate over current buffer to convert already stored Buffers:
-  let p = this._readableState.buffer.head;
   let content = '';
-  while (p !== null) {
-    content += decoder.write(p.data);
-    p = p.next;
+  for (const data of this._readableState.buffer) {
```

Comment:
Is this as fast as the original?

Reply:
I don't know... trying to figure out the whole benchmark thing at the moment...

Reply:
I don't know... trying to figure out the whole benchmark thing at the moment

Reply:
```streams/readable-readall.js n=200                -0.94 %       ±2.54% ±3.38% ±4.40%```